### PR TITLE
Update .opam file to use with-dev-setup instead of with-test

### PR DESCRIPTION
**Pull request by feihong**
Addressing https://github.com/melange-re/melange-re.github.io/pull/200/files#r1811990250

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "melange-opam-template",
   "scripts": {
     "preinstall-opam": "opam update",
-    "install-opam": "opam install -y . --deps-only --with-test",
+    "install-opam": "opam install -y . --deps-only --with-test --with-dev-setup",
```
[the remark this answers is not in the document]
Comment (jchavarri) [reply package.json:5]: Yes, please.

[the remark this answers is not in the document]
Comment (feihong) [reply package.json]: Fixed, merging now

**Comment by jchavarri** [comment]
CI is [failing](https://github.com/melange-re/melange-opam-template/actions/runs/11520981914/job/32073567152#step:5:47):

```
opam: unknown option `--with-dev-setup', did you mean `-w' ?
```

we should fix this before merging.

**Comment by feihong** [comment]
@jchavarri Fixed by upgrading ocaml/setup-ocaml action from v2 to v3 so that it uses opam 2.2: https://github.com/melange-re/melange-opam-template/pull/38/commits/2d2b43c6a1d2d58d8a84a50fc53732ad129ede29

**Comment by jchavarri** [comment]
Thanks!

**Pull request merged**